test/integration/client-side-encryption/driver.test.ts

```diff
@@ -215,10 +209,86 @@ describe('Client Side Encryption Functional', function () {
         const expected = BSON.deserialize(BSON.serialize(data, bsonOptions), bsonOptions);
 
         const coll = this.encryptedClient.db(dataDbName).collection(dataCollName);
-        return Promise.resolve()
-          .then(() => coll.insertOne(data, bsonOptions))
-          .then(result => coll.findOne({ _id: result.insertedId }, bsonOptions))
-          .then(actual => expect(actual).to.containSubset(expected));
+        const result = await coll.insertOne(data, bsonOptions);
+        const actual = await coll.findOne({ _id: result.insertedId }, bsonOptions);
+        const gValue = actual.g;
+        delete actual.g;
+
+        expect(actual).to.deep.equal(expected);
+        expect(gValue).to.equal(bsonOptions.ignoreUndefined ? data.g : null);
+      });
+    }
+  });
+
+  describe('key order aware command properties', () => {
+    let client: MongoClient;
+    let collection: Collection;
+
+    beforeEach(async function () {
+      if (this.configuration.clientSideEncryption == null) {
+        return;
+      }
+      const encryptionOptions = {
+        monitorCommands: true,
+        autoEncryption: {
+          keyVaultNamespace,
+          kmsProviders: { local: { key: 'A'.repeat(128) } }
+        }
+      };
+      client = this.configuration.newClient({}, encryptionOptions);
+      collection = client.db(dataDbName).collection('keyOrder');
+    });
+
+    afterEach(async () => {
+      if (client) await client.close();
+    });
+
+    describe('find', () => {
+      it('should maintain ordered sort', metadata, async function () {
+        const events: CommandStartedEvent[] = [];
+        client.on('commandStarted', ev => events.push(ev));
+        const sort: [string, SortDirection][] = [
```

Comment:
should we freeze this object to make sure it's not being mutated (same in the other test cases in this file)? otherwise the event equality check is less meaningful

Reply:
Done!

Reply:
There's 2 more cases